aio.core/aio/core/functional/generator.py

```diff
@@ -0,0 +1,62 @@
+
+from typing import AsyncIterable, AsyncGenerator, Callable, Coroutine, Iterable
+
+from aio.core import functional
+
+
+class AwaitableGenerator:
+    """Wrap an asynchronous generator as optionally awaitable.
+
+    For example, you may have a low level async function that generates
+    results, but where in some cases you want to just wait for the gathered
+    results.
+
+    The 2 methods of handling results show below are functionally equivalent,
+    save for the that the in the second example all results are generated
+    before iteration begins.
+
+    ```python
+
+    async def some_generator():
+        for x in range(0, 5):
+            yield x
+
+    async for result in AwaitableGenerator(some_generator()):
+        do_something_with(result)
+
+    for result in await AwaitableGenerator(some_generator()):
+        do_something_with(result)
+
+    ```
+
+    By default results will be collected into a `list`, using the
+    `async_list` function. You can specify a custom collector as
```

Comment:
swap the default to a `set` - assuming the yield order of the underlying generator is indeterminate, we should only want a list if we expressly want the order that they were returned in

Reply:
hmm - scrap that we need a list because we want all of the results